src/party.cpp

```diff
@@ -399,6 +399,8 @@ bool Party::setSharedExperience(Player* player, bool sharedExpActive)
 
 		if (this->sharedExpEnabled) {
 			leader->sendTextMessage(MESSAGE_INFO_DESCR, "Shared Experience is now active.");
+		} else if (!this->hasRequiredLevelToShare) {
```

Comment:
Here, you should call `canUseSharedExperience` instead of setting a variable. It's not that costly anyway. This way you wouldn't need to change the constness of the function.